File: integrations/src/traverse/java/juuxel/adorn/compat/traverse/TraverseCompat.java
package juuxel.adorn.compat.traverse;

import com.google.auto.service.AutoService;
import juuxel.adorn.block.variant.BlockVariant;
import juuxel.adorn.block.variant.BlockVariantSet;

import java.util.List;

@AutoService(BlockVariantSet.class)
public final class TraverseCompat implements BlockVariantSet {
    @Override
    public List<BlockVariant> getWoodVariants() {
        return List.of(new BlockVariant.Wood("traverse/fir"));
    }
}

File: src/main/kotlin/juuxel/adorn/compat/TraverseCompat.kt
package juuxel.adorn.compat

import juuxel.adorn.Adorn
import juuxel.adorn.api.block.AdornBlockBuilder
import juuxel.adorn.api.block.BlockVariant

object TraverseCompat {
    fun init() {
        // TODO: Get rid of this in 1.17
        val oldFir = BlockVariant.Wood("traverse_fir")
        AdornBlockBuilder.create(oldFir)
            .withPost()
            .withPlatform()
            .withStep()
            .withDrawer()
            .withChair()
            .withTable()
            .withKitchenBlocks()
            .withShelf()
            .withCoffeeTable()
            .registerIn(Adorn.NAMESPACE)

        val newFir = BlockVariant.Wood("traverse/fir")
        AdornBlockBuilder.create(oldFir)
            .withBench()
            .registerIn(Adorn.NAMESPACE)
    }
}
